### ISSUE TITLE
quantify_output: handle same sample names as sample sheet

### ISSUE THREAD

**Issue by samnooij** [DennisSchmitz/Jovian_archive#59]
@kroonma updated the sample_sheet generator to also work with the ENA filename scheme `[filename]_1.fastq` and with regular '[filename]_R1.fastq' naming scheme. See here `fq_pattern = re.compile("(.*)_R?(1|2).*\.f(ast)?q(\.gz)?")` 

Can this be updated to work with the sample_sheet? Or to use a similar regex as listed above.

_Originally posted by @DennisSchmitz in https://github.com/DennisSchmitz/Jovian/issues/57#issuecomment-495636345_

---

The way `bin/quantify_profiles.py` parses sample names was simply to cut the last 3 characters of (assuming they would be '_R1'). In a developmental version, this has been changed to use RegEx to find '_R1' and anything behind it and replace it with nothing.
This, however, does not take into account the possibility of sample names with the 'R' in them.

Therefore, the script needs to change the way sample names are parsed. Perhaps they can be taken from the sample sheet, or the same RegEx can be used.

P.S. This happens when reading the multiqc output files of fastqc and trimmomatic. In there, I am using these suffices to extract the necessary information and reconstruct sample names. Perhaps I can find a less error-prone method of working with sample names here.

**Comment by samnooij** [comment]
Most problems should be solved with https://github.com/DennisSchmitz/Jovian/commit/6ff03ab6560faf665b260b8c3f65194b2d5ff468. Issues may still occur with complicated sample names with multiple '_1' in them. I am leaving this issue open because I think this part of the script can still be improved.

**Comment by kroonma** [comment]
@samnooij did the [modified regex](https://regexr.com/4etjf) not work?

When I previously looked at the code I didn't realise it was a Pandas dataframe. It seems it accepts regular expressions directly:
```python
read_nrs.Sample = read_nrs.Sample.str.replace(r'_R?(1|2)[^R\s]*$', '')
```
No time a.t.m. to test and make a PR out of this.

**Comment by DennisSchmitz** [comment]
Sam, I fixed #42 in 7aa5cdb and I think the problem is similar. Might be useful to check it out.

**Comment by DennisSchmitz** [comment]
This issue is fixed in the `dev` branch, will be included in the next release. Closing it.
